modal/functions.py

```diff
@@ -1261,6 +1262,7 @@ def _hydrate_metadata(self, metadata: Optional[Message]):
         self._use_function_id = metadata.use_function_id
         self._use_method_name = metadata.use_method_name
         self._class_parameter_info = metadata.class_parameter_info
+        self._method_handle_metadata = dict(metadata.method_handle_metadata)
```

Comment:
Might be cleaner at this point to just store the full `metadata` object on self - especially if we move out a bunch of the class-specific stuff to a separate non-_Object `MethodHandle` class - since most of these attributes would then never be used by _Function itself.

Could do that when we create that class and move out the `_bind_method()` to a constructor of that class instead. Just a thought